aten/src/ATen/SparseGCSTensorImpl.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <ATen/Tensor.h>
+#include <ATen/TensorUtils.h>
+#include <ATen/SparseTensorUtils.h>
+#include <ATen/SparseCOOTensorImpl.h>
+#include <c10/core/TensorImpl.h>
+#include <c10/util/Exception.h>
+
+namespace at {
+  
+// TODO: since many methods in SparseCOOTensorImpl can be used by GCS sparse tensor directly
+// we probably should have some superclass between TensorImpl and GCSTensorImpl that is
+// shared between COO and GCS tensors.
+struct CAFFE2_API SparseGCSTensorImpl : public TensorImpl {
+  Tensor pointers_;
```

Comment:
I probably wouldn't call this pointers, that suggests you're actually storing pointers in the tensor, which I'm pretty sure you're not doing.